cli2/src/Morphir/Elm/CLI.elm

```diff
@@ -155,15 +182,40 @@ process msg =
             let
                 decodeInput : Decode.Decoder BuildIncrementallyInput
                 decodeInput =
-                    Decode.map4 BuildIncrementallyInput
+                    Decode.map5 BuildIncrementallyInput
                         (Decode.field "options" FrontendCodec.decodeOptions)
                         (Decode.field "packageInfo" FrontendCodec.decodePackageInfo)
                         (Decode.field "fileChanges" FileChangesCodec.decodeFileChanges)
+                        (Decode.field "dependencies" decodeDependenciesInput)
                         (Decode.field "distribution" DistroCodec.decodeVersionedDistribution)
+
+                decodeDependenciesInput : Decode.Decoder (List Distribution)
+                decodeDependenciesInput =
+                    Decode.list DistroCodec.decodeVersionedDistribution
+
+                insertDependenciesIntoDistro : List Distribution -> Distribution -> Distribution
+                insertDependenciesIntoDistro dependencyList distribution =
+                    dependencyList
+                        |> List.foldl
+                            (\distro distroResult ->
+                                let
+                                    dependencySpec : Specification ()
+                                    dependencySpec =
+                                        lookupPackageSpecification distro
+
+                                    dependencyName : PackageName
+                                    dependencyName =
+                                        lookupPackageName distro
+                                in
+                                distroResult
+                                    |> Distribution.insertDependency dependencyName dependencySpec
+                            )
+                            distribution
```

Comment:
This section seems to be duplicated above. Can you move it out to a function and call that from both places?